src/modules/cmdpal/ext/Microsoft.CmdPal.Ext.Bookmark/Helpers/BookmarkTypeHelper.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CmdPal.Ext.Bookmarks.Models;
+
+namespace Microsoft.CmdPal.Ext.Bookmarks.Helpers;
+
+public static partial class BookmarkTypeHelper
+{
+    /*
+     * Summary:
+     * If bookmark has a space, we assume it's a shell command. eg: "python test.py" or "test.ps1 /C /D"
+     * Ok fine, we can ensure the bookmark don't have spaces now.
+     * So:
+     * 1. Check if it's a valid url.
+     * 2. Check if it's a existing folder.
+     * 3. if it's a existing file, it also have the possibility to be a shell command file. eg: "test.cmd" or "test.ps1". So, check the extension. If not, assume it's a normal file.
+     * By default, we assume it's Web Link.
+     */
+
+    public static BookmarkType GetBookmarkTypeFromValue(string bookmark)
+    {
+        var splittedBookmarkValue = bookmark.Split(" ");
+
+        if (splittedBookmarkValue.Length > 1)
+        {
+            // absolutely it's a shell command
+            // we don't need to check the file name
+            var executableFileName = splittedBookmarkValue[0];
+            var executableExtension = System.IO.Path.GetExtension(executableFileName);
+
+            // if it's a cmd
+            if (executableExtension == ".cmd" || executableExtension == ".bat")
+            {
+                return BookmarkType.Cmd;
+            }
+
+            // Otherwise, we assume it's a powershell or pwsh.
+            // Prefer to use pwsh, but check if pwsh is installed
+            // if not, we use powershell
+            if (EnvironmentsCache.Instance.TryGetExecutableFileFullPath("pwsh.exe", out _))
```

Comment:
isn't this force powershell 5 scripts to run in pwsh 7 if both are installed? Bad idea.

Reply:
no. don't do this. Because they are not compatible. Well you can decide based on executable used. or not?